view/frontend/web/js/view/payment/method-renderer/adyen-apple-pay-method.js

```diff
@@ -0,0 +1,184 @@
+/*
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2017 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+define(
+    [
+        'jquery',
+        'Magento_Checkout/js/model/quote',
+        'Magento_Checkout/js/view/payment/default',
+        'Magento_Checkout/js/action/place-order',
+        'Magento_Checkout/js/model/payment/additional-validators',
+        'Magento_Checkout/js/model/url-builder',
+        'mage/storage',
+        'mage/url',
+        'Magento_Ui/js/model/messages'
+    ],
+    function ($, quote, Component, placeOrderAction, additionalValidators, urlBuilder, storage, url, Messages) {
+        'use strict';
+        return Component.extend({
+            self: this,
+            defaults: {
+                template: 'Adyen_Payment/payment/apple-pay-form'
+            },
+
+            /**
+             * @returns {Boolean}
+             */
+            isShowLegend: function () {
+                return true;
+            },
+            setPlaceOrderHandler: function (handler) {
+                this.placeOrderHandler = handler;
+            },
+            setValidateHandler: function (handler) {
+                this.validateHandler = handler;
+            },
+            getCode: function () {
+                return 'adyen_apple_pay';
+            },
+            getData: function () {
+                return {
+                    'method': this.item.method,
+                    'additional_data': {}
+                };
+            },
+            isActive: function () {
+                return true;
+            },
+            /**
+             * @override
+             */
+            placeApplePayOrder: function (data, event) {
+                event.preventDefault();
+                var self = this;
+                if (!additionalValidators.validate()){
+                    return false;
+                }
+                var request = {
+                    countryCode: 'US',
+                    currencyCode: 'EUR',
+                    supportedNetworks: ['visa', 'masterCard', 'amex', 'discover'],
```

Comment:
shall we use DL results to populate these variants?

Reply:
Let's think about this one how to do best. It could be that DL is not enabled. We could make this an option in apple pay section, we need to do this as well for M1.